src/Chapter2.hs

```diff
@@ -620,7 +625,26 @@ Write a function that takes elements of a list only on even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
+takeEven' :: [a] -> [a]
+takeEven' list = go list []
+  where
+    go :: [a] -> [a] -> [a]
+    go list newList = 
+      case list of
+        [] -> reverse newList
+        (x : []) -> reverse (x : newList)
+        (x1 : x2 : xs) -> go xs (x1 : newList)
+         
+takeEven :: [a] -> [a]
```

Comment:
First, I was trying to reverse the passed list and then call go function on it but it works incorrectly if the list has two elements. Instead of returning the first element, it would return the second one.
So I reverse the list constructed in go function. Otherwise I would have to prepend to newList N / 2 times where N is list length. Adding elements to the end of a list is an expensive operation so this is unwanted. Reversing a list isn't cheap either... I guess it's still better to reverse the new list. Right?

Reply:
I made a comment about this approach above, but for this question, I would say yes, though it is already quite expensive to use `length` as well 🙂